src/Microsoft.Performance.SDK.Runtime/CustomDataSourceReference.cs

```diff
@@ -207,314 +343,133 @@ protected CustomDataSourceReference(CustomDataSourceReference other)
-            }
-
-            /// <inheritdoc />
-            public override CustomDataSourceReference CloneT()
-            {
-                this.ThrowIfDisposed();
-                return new CustomDataSourceReferenceImpl(this);
-            }
-
-            /// <inheritdoc />
-            public override bool Equals(object obj)
-            {
-                this.ThrowIfDisposed();
-                if (obj is null)
-                {
-                    return false;
-                }
-
-                if (ReferenceEquals(this, obj))
-                {
-                    return true;
-                }
-
-                var other = obj as CustomDataSourceReferenceImpl;
-
-                return other != null &&
-                    this.Name.Equals(other.Name) &&
-                    this.Guid.Equals(other.Guid) &&
-                    this.Description.Equals(other.Description) &&
-                    (this.dataSourceAttributes.IsSubsetOf(other.dataSourceAttributes) &&
-                     other.dataSourceAttributes.IsSubsetOf(this.dataSourceAttributes));
+                this.createdProcessors = null;
+                this.commandLineOptionsRO = null;
+                this.dataSourceAttributes = null;
+                this.description = null;
+                this.guid = default;
+                this.name = null;
             }
 
-            /// <inheritdoc />
-            public override int GetHashCode()
-            {
-                this.ThrowIfDisposed();
-                unchecked
-                {
-                    var hash = base.GetHashCode();
-
-                    hash = ((hash << 5) + hash) ^ this.Name.GetHashCode();
-                    hash = ((hash << 5) + hash) ^ this.Guid.GetHashCode();
-                    hash = ((hash << 5) + hash) ^ this.Description.GetHashCode();
-                    hash = ((hash << 5) + hash) ^ this.DataSources.GetHashCode();
-
-                    return hash;
-                }
-            }
-
-            /// <inheritdoc />
-            public override string ToString()
-            {
-                this.ThrowIfDisposed();
-                return $"{this.Name} - {this.Guid} ({this.AssemblyPath})";
-            }
-
-            /// <inheritdoc />
-            protected override void Dispose(bool disposing)
-            {
-                if (this.isImplDisposed)
-                {
-                    return;
-                }
-
-                if (disposing)
-                {
-                    foreach (var p in this.createdProcessors)
-                    {
-                        try
-                        {
-                            this.Instance.DisposeProcessor(p);
-                        }
-                        catch
-                        {
-                        }
+            base.Dispose(disposing);
+            this.isDisposed = true;
+        }
 
-                        p.TryDispose();
-                    }
+        private void InitializeThis(
```

Comment:
since we can't do a ctor for this initialization, should we add a bool `this.isInitialized` that gets set to `true` in this method? Then anywhere we do `ThrowIfDisposed` we do a `ThowIfNotInitialized`. Just to make sure we actually call this

Reply:
sure. That should probably be an assert as it is a bug in the runtime to not call it, and the caller has no control over that at all.